tests/utils/utils.go

```diff
@@ -0,0 +1,340 @@
+		defer client.Close()
+
+		// Loop until each node has advanced to >= the height of the block that emitted the warp log
+		for {
+			block, err := client.BlockByNumber(cctx, nil)
+			Expect(err).Should(BeNil())
+			if block.NumberU64() >= height {
+				log.Info("client accepted the block containing SendWarpMessage", "client", i, "height", block.NumberU64())
+				break
+			}
+		}
+	}
+}
+
+// Constructs the aggregate signature, packs the Teleporter message, and relays to the destination
+func RelayMessage(
+	ctx context.Context,
+	sourceBlockHash common.Hash,
+	sourceBlockNumber *big.Int,
+	source SubnetTestInfo,
+	destination SubnetTestInfo,
+) {
+	log.Info("Fetching relevant warp logs from the newly produced block")
+	logs, err := source.ChainWSClient.FilterLogs(ctx, interfaces.FilterQuery{
+		BlockHash: &sourceBlockHash,
+		Addresses: []common.Address{warp.Module.Address},
+	})
+	Expect(err).Should(BeNil())
+	Expect(len(logs)).Should(Equal(1))
+
+	// Check for relevant warp log from subscription and ensure that it matches
+	// the log extracted from the last block.
+	txLog := logs[0]
+	log.Info("Parsing logData as unsigned warp message")
+	unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+	Expect(err).Should(BeNil())
+
+	// Set local variables for the duration of the test
+	unsignedWarpMessageID := unsignedMsg.ID()
+	unsignedWarpMsg := unsignedMsg
+	log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+	// Loop over each client on chain A to ensure they all have time to accept the block.
+	// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+	// has accepted the block.
+	WaitForAllValidatorsToAcceptBlock(ctx, source.ChainNodeURIs, source.BlockchainID, sourceBlockNumber.Uint64())
+
+	// Get the aggregate signature for the Warp message
+	log.Info("Fetching aggregate signature from the source chain validators")
+	warpClient, err := warpBackend.NewClient(source.ChainNodeURIs[0], source.BlockchainID.String())
+	Expect(err).Should(BeNil())
+	signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
+	Expect(err).Should(BeNil())
+
+	// Construct the transaction to send the Warp message to the destination chain
+	signedTx := CreateReceiveCrossChainMessageTransaction(
+		ctx,
+		signedWarpMessageBytes,
+		big.NewInt(1),
+		teleporterContractAddress,
+		fundedAddress,
+		fundedKey,
+		destination.ChainWSClient,
+		destination.ChainIDInt,
+	)
+
+	log.Info("Sending transaction to destination chain")
+	receipt := SendTransactionAndWaitForAcceptance(ctx, destination.ChainWSClient, signedTx)
+
+	receiveCrossChainMessageLog := receipt.Logs[0]
+	var event ReceiveCrossChainMessageEvent
+	err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&event, "ReceiveCrossChainMessage", receiveCrossChainMessageLog.Data)
+	Expect(err).Should(BeNil())
```

Comment:
I think we should eitehr return the `receipt` or `event`, then in the `BasicOneWaySend` test make sure that the received message has Teleporter fields that matches the `teleporterMessage` that was sent

Reply:
I modified `RelayMessage` to return the receipt, but I think for this test it's sufficient to just check against the `TeleporterMessenger` contract by calling `messageReceived`. Other tests may want to verify the fields directly though, so good call to have this function return the receipt.

Reply:
are you thinking a separate test case to verify we sent and received the correct teleporter message with correct fields? I was thinking this test case would include sending, check received, and check that we received correct message.

Reply:
More specifically I think we can return the receipt and parse the receipt log for receive event in `BasicOneWaySend)`
```
	receipt = utils.RelayMessage(ctx, receipt.BlockHash, receipt.BlockNumber, subnetAInfo, subnetBInfo)
	destinationBind, err := teleportermessenger.NewTeleportermessenger(teleporterContractAddress, subnetBInfo.ChainWSClient)
	Expect(err).Should(BeNil())
	receiveEvent, err := destinationBind.ParseReceiveCrossChainMessage(*receipt.Logs[0])
	Expect(err).Should(BeNil())
	Expect(receiveEvent.Message.MessageID).Should(Equal(teleporterMessageID))
	Expect(receiveEvent.OriginChainID).Should(Equal(subnetAInfo.BlockchainID))
	Expect(receiveEvent.Message).Should(Equal(sendCrossChainMessageInput.Message))
```

Reply:
This PR is to set up the testing framework, not to implement robust test cases. The test case here is the minimum test case to demonstrate the framework works as intended for this use case. Let's plan to revisit `basic_one_way_send` (or remove it altogether) in a future PR.